src/ratelimit_manager.cpp

```diff
@@ -491,4 +501,55 @@ time_t RateLimitManager::get_current_time() {
 
 void RateLimitManager::_set_base_timestamp(const time_t& timestamp) {
     base_timestamp = timestamp;
+}
+
+
+const nlohmann::json RateLimitManager::get_all_throttled_entities_json() {
+    nlohmann::json throttled_entities_array = nlohmann::json::object();
```

Comment:
No lock? Also go through all other methods in rate limit manager to ensure that locks are not missed.